src/lib/Util/FieldDefinitionGroupsUtil.php

```diff
@@ -30,12 +30,21 @@ public function __construct(FieldsGroupsList $fieldsGroupsListHelper)
      */
     public function groupFieldDefinitions(iterable $fieldDefinitions): array
     {
+        $fieldDefinitionsGroups = [];
+        foreach ($fieldDefinitions as $fieldDefinition) {
+            $fieldDefinitionsGroups[] = $fieldDefinition->fieldGroup;
+        }
```

Comment:
Why not start preparing the data structure here immediately? You will be iterating over the same data again in a moment.

This should make the code simpler. I understand that what you're trying to do is to start creating the array here, so it will retain the order.

Additionally, since we're working on an `iterable`, we can't be sure we are even allowed to iterate over the results more than once (`Generator` exhaustion issue).